WIP User Guide

JuliaPoo opened the pull request:
This pull mostly:

1. Fix factual errors in the UG
2. Established a format for the documentation of each command
    - E.g., with regards to communicating the format of each command, just modify the LATEX given for `edit` or `add`.
4. Restructured the Features section.

The old UG is in the file `UserGuide.md.old`

There are still many issues to be fixed. Refer to #83.

Currently the User Guide looks like [this](https://github.com/AY2324S2-CS2103T-W11-2/tp/files/14967675/CCA.Manager.-.User.Guide.-.AddressBook.Level-3.pdf). There is still a lot to be done, especially regarding documenting each of the commands to a satisfactory standard.

The absolutely most time consuming thing right now, is to document each command properly. Please refer to `edit` and `add` where I've partially done up the documentation (still missing potential failures section for each), as a reference for documenting other commands.

Furthermore, there are still a few missing sections. Refer to the notes in #83. E.g., a section with an explanation of the GUI elements of the UG itself.

Furthermore, there are definitely small things to take note of, such as broken links, and a more informal language.

Fidget-Spinner commented on docs/UserGuide.md:
```diff
@@ -108,268 +122,397 @@ To run a command, type the command in the **Command Box** and press Enter to exe
-* You can remove all the person’s roles by typing `r/` without
-    specifying any roles after it.
+#### Track a person owing money: `owe`
+Set up owe amount of money in each person.
```
```suggestion
Set up amount of money each person owes
```
Also dropped the period because the rest don't have it.

Fidget-Spinner reviewed (commented):
One minor comment, else LGTM.

JuliaPoo commented:
In addition to fixing checkstyle, I added a broken links detector:
![image](https://github.com/AY2324S2-CS2103T-W11-2/tp/assets/57632293/6d7c861b-22dd-437a-b6e4-4eff5a0e1f3e)

by appending the following code at the bottom of the page:

```html
<script>
let meow = [...document.getElementsByTagName("a")].filter(a => a.href.includes("#")).filter(a => document.getElementById(a.href.split("#")[1]) === null).map(a => `  -- [${a.textContent}](${a.href.split("#")[1]})`).join("\n");
if (meow !== "") alert(`Broken links:\n${meow}`)
</script>
```
